internal/provider/template_resource.go

```diff
@@ -910,19 +928,6 @@ func (a *activeVersionValidator) ValidateList(ctx context.Context, req validator
 	if !active {
 		resp.Diagnostics.AddError("Client Error", "At least one template version must be active.")
 	}
-
-	// Check all versions have unique names
-	uniqueNames := make(map[string]struct{})
-	for _, version := range data {
-		if version.Name.IsNull() {
-			continue
-		}
-		if _, ok := uniqueNames[version.Name.ValueString()]; ok {
```

Comment:
If two names were populated using Terraform variables, they'd be `Unknown` and `ValueString` would return the empty strin for both, which would then lead to a false duplicate in the map.